skfp/model_selection/splitters/pubchem_split.py

```diff
@@ -0,0 +1,456 @@
+    the newest ones assigned to the test subset and the rest to the training subset.
+
+    In case of no data available (no literature or no PubChem entry) the behavior
+    is governed by the ``not_found_behavior`` parameter. Full DataFrame with columns
+    ``["SMILES", "split", "year"]`` can also be returned.
+
+    Note that this split can be very slow for large number of molecules, since requests
+    to PubChem frequently fail due to the "servers overloaded" error.
+
+    The split fractions (train_size, test_size) must sum to 1.
+
+    Parameters
+    ----------
+    data : sequence
+        A sequence representing either SMILES strings or RDKit ``Mol`` objects.
+
+    additional_data: list[sequence]
+        Additional sequences to be split alongside the main data (e.g., labels or feature vectors).
+
+    train_size : float, default=None
+        The fraction of data to be used for the train subset. If None, it is set to 1 - test_size.
+        If test_size is also None, it will be set to 0.8.
+
+    test_size : float, default=None
+        The fraction of data to be used for the test subset. If None, it is set to 1 - train_size.
+        If train_size is also None, it will be set to 0.2.
+
+    not_found_behavior : {"train", "test", "remove"}, default="test"
+        What to do with molecules not found in PubChem, or if molecule does not have
+        any literature data available. Such molecules can be put into train set,
+        test set, or removed altogether.
+
+    return_indices : bool, default=False
+        Whether the method should return the input object subsets, i.e. SMILES strings
+        or RDKit ``Mol`` objects, or only the indices of the subsets instead of the data.
+
+    n_jobs : int, default=5
+        The number of jobs to run in parallel for fetching data from PubChem. Note that
+        since those are requests relying on I/O, threads are used instead of processes.
+        At most 5 requests per second are made to avoid throttling, which limits
+        parallelism even if high ``n_jobs`` is set.
+
+    Returns
+    ----------
+    subsets : tuple[list, list, ...]
+    Tuple with train-test subsets of provided arrays. First two are lists of SMILES
+    strings or RDKit `Mol` objects, depending on the input type. If `return_indices`
+    is True, lists of indices are returned instead of actual data.
+
+    References
+    ----------
+    .. [1] `Kim S, Thiessen PA, Cheng T, Yu B, Bolton EE.
+        "An update on PUG-REST: RESTful interface for programmatic access to PubChem."
+        Nucleic Acids Res. 2018 Jul 2;46(W1):W563-W570.
+        <https://doi.org/10.1093/nar/gky294>`_
+    """
+    years = _get_pubchem_years(data, n_jobs)
+    data_df = pd.DataFrame({"idx": list(range(len(data))), "year": years})
+
+    if not_found_behavior != "remove":
+        not_found_mols = list(data_df[data_df["year"].isna()].idx)
+
+    data_df = data_df[~data_df["year"].isna()]
```

Comment:
This is controlled by `not_found_behavior`. This line should only happen for `remove`. Note that `.sort_values` takes `na_position` argument, which will be useful for `train` and `test` values.